### ISSUE TITLE
Add an environment variable to remove the progress bar for validating & signing the add on with the web-ext sign command

### ISSUE THREAD

**Issue by hmx789** [mozilla/web-ext#2154]
### Is this a feature request or a bug?
This is a feature request.

### What is the current behavior?
Currently when we run web-ext sign command there is a progress bar that for validating the add on and signing the addon(added images below). Don't get me wrong, the progress bar is very cool on my local terminal but on CI this causes unnecessary logging.
![image](https://user-images.githubusercontent.com/36273963/107573521-3f490480-6bb3-11eb-8e2a-5612ce69d5d7.png)
![image](https://user-images.githubusercontent.com/36273963/107573550-47a13f80-6bb3-11eb-8f2a-3a747d3a4e88.png)
![image](https://user-images.githubusercontent.com/36273963/107573644-69022b80-6bb3-11eb-8706-93416afd4860.png)

### What is the expected or desired behavior?
I would like it if there was an environment variable that I could pass into the command to remove the progress updates. I would only like to remove the progress bar updates. I believe it is still valuable for us to provide the validation results link, the output of downloaded file and the Extension ID.

**Comment by Rob--W** [comment]
Patches are welcome. To resolve this issue, the `sign-addon` dependency needs to be updated to support this (see https://github.com/mozilla/sign-addon/issues/710 for details), and then web-ext needs to account for it.

If it is possible to automatically detect the scenario (e.g. by using `isTTY` to detect interactive output), then we can automatically detect and fix it so that `web-ext` users don't need special configuration. Otherwise a new CLI option or environment variable would be acceptable too.

**Comment by manthony9** [comment]
Hey @Rob--W  can I take this issue up? I've been in conversation with @willdurand and I think I have some headway. Would need a little assistance to go further.

**Comment by willdurand** [comment]
> Hey @Rob--W can I take this issue up? I've been in conversation with @willdurand and I think I have some headway. Would need a little assistance to go further.

@manthony9 FWIW, we need to update `sign-addon` first before being able to move forward with _this_ issue.

**Comment by manthony9** [comment]
Oh! my bad...let me work on sign-add on first..

**Comment by Rob--W** [comment]
@willdurand When you're back, can you reply to https://github.com/mozilla/sign-addon/pull/775 ?

**Comment by willdurand** [comment]
`sign-addon` v3.7 now provides a option to disable the pseudo progress bar: https://github.com/mozilla/sign-addon/releases/tag/3.7.0

**Comment by rpl** [comment]
@manthony9 the change in sign addon made it to v3.7, but there are still changes on the web-ext side to leverage that new sign-addon option, are you interested to look into contributing that part?

**Comment by manthony9** [comment]
ill be happy to contribute.

**Comment by manthony9** [comment]
@rpl would you be assigning the issue to me or should I start working on it as it is. I'm sorry I am not sure what the best approach is.

**Comment by rpl** [comment]
> @rpl would you be assigning the issue to me or should I start working on it as it is. I'm sorry I am not sure what the best approach is.

@manthony9 we assign mentored issues once a pull request has been opened, and so feel free start working on it.

Nevertheless, feel free to ask us questions if you need some more guidance about the work needed to fix this issue (@Rob--W is the mentor assigned to this bug).

**Comment by manthony9** [comment]
Hi @Rob--W , 

I am trying to get to see the [https://github.com/mozilla/sign-addon/pull/826](url) update working. I've got the web-ext set-up and added the disableProgressBar param as per @rpl 's  comment, setting it to true, but I still see the progress bar. I'm sure there's something I am doing wrong. Could you share your thoughts on the same.

![image](https://user-images.githubusercontent.com/13238071/140250625-c6caaa3c-3efe-4c09-8d76-91a8f2f01ac4.png)

**Comment by rpl** [comment]
@manthony9 what are the steps that you are using to test that change?

I tried to add `disableProgressBar: true` locally and it seems to be disabling the progress bar as expected for me.

There is a script in tests/functional/fake-amo-server.js which is used in the functional test file named `tests/functional/test.cli.sign.js` to run a fake AMO server to be used for testing that the `web-ext sign` does work as expected (more as a smoke test, it is not as detailed as the unit tests, but it does use the web-ext bin script with all the real dependencies as the user would do).

That script is limited (the purpose is just to use it to validate a single submission triggered by `web-ext sign`), but it is useful to get a quick feedback. 

The following how I tested the change:
- first of all: make sure to pull the last changes available in this repository and to update all the dependencies available in node_modules/ (`git pull ...` + `npm ci`)
- then test without the additional disableProcessBar option set to true first (to confirm where the progress bar will be in the output):
  - ensure to build web-ext src into the bundle stored in dist (`npm run build`)
  - in one shell run the fake amo server: ./tests/functional/fake-amo-server.js
  - in another shell run `web-ext sign` with the following cli options: `./bin/web-ext sign -s path/to/any-test-extension --api-url-prefix http://localhost:8989/fake/api/v4 --api-key FAKEAPIKEY --api-secret FAKEAPISECRET` 
    - (if you don't have a test extension, you can get one from https://github.com/mdn/webextensions-examples or create one on the fly using `npx create-webextension test-ext-webext-sign-progressbar`)
  - Actual (and expected) behavior:
    - the progress bar was part of the output is logged in the console
```
Applying config file: ./...
Building web extension from /...
Validating add-on [.........................................................................................................................]
Validation results: undefined
Signing add-on [............................................................................................................................]
Downloading signed files: ...
Downloaded:
    ./....xpi
Extension ID: an-addon-guid
SUCCESS
```

- then to test with the additional disableProcessBar option set to true first (to confirm that the progress bar is gone from the output):
  - apply the needed changes to src/cmd/sign.js
  - make sure to rebuild the bundled file so that it will include the changes (`npm run build`)
  - on one of your shell: stop and run again the fake-amo-server.js script
  - in another shell run the same `./bin/web-ext sign ...` command with the same options as for the previous test
  - Actual (and expected behavior):    
    - the progress bar was NOT part of the output is logged in the console
```
Applying config file: ./...
Building web extension from /...
Validation results: undefined
Downloading signed files: ...
Downloaded:
    ./....xpi
Extension ID: an-addon-guid
SUCCESS
```

Give it a try to this steps and let us know how it goes, if you are still unable to reproduce the behavior described above, it may be useful to also:
- submit a PR with all the changes you have locally
- and a description of the steps you have been using to test it out

Basically everything that may allow us to more easily see what may differs between what I tried and described above and what you did locally.

**Comment by manthony9** [comment]
@rpl thank you so much for that detailed breakdown. Able to run it with the disableProgressbar now. Starting work on the fix. :)
